object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/FrameworkSupport.java

```diff
@@ -0,0 +1,38 @@
+package org.checkerframework.checker.objectconstruction.framework;
+
+import com.sun.source.tree.NewClassTree;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/** add CalledMethods annotations to code generated by framework */
+public interface FrameworkSupport {
+
+  /**
+   * Allows for supporting a framework's {@code toBuilder} routine. Typically, the returned Builder
+   * setters invoked. So, the framework support can add a {@link
+   * org.checkerframework.checker.objectconstruction.qual.CalledMethods} annotation capturing this
+   * fact.
+   *
+   * @param t method named "toBuilder"
+   */
+  public void handleToBuilder(AnnotatedTypeMirror.AnnotatedExecutableType t);
```

Comment:
I think the code in `ObjectConstructionAnnotatedTypeFactory` actually could pass any method here? But the Javadoc says that the method must be named "toBuilder". Please make the contract more clear.